### PR TITLE
integration with psalm

### PR DESCRIPTION
**Pull request by pgichuki**
#977

### CODE REVIEW

**Comment by pgichuki** [comment]
To initilize;

1. `docker exec -it herams-backend_phpfpm_1 /bin/sh`
2. `/project/vendor/vimeo/psalm/psalm --init --root=project`

To config PhpStorm for code inspection [Configure a Psalm script associated with a PHP interpreter]
(https://www.jetbrains.com/help/phpstorm/using-psalm.html#configure-php-code-sniffer-script-associated-with-php-interpreter)

**Comment by SamMousa** [comment]
Initialization should not be needed by everyone, just one person does it and you commit the shared configuration to the repo.

File: composer.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
         "kartik-v/yii2-widget-select2": "^2.1",
         "kodus/file-cache": "^1.1",
         "kartik-v/yii2-widget-typeahead": "^1.0",
-        "league/iso3166": "^2.1",
+        "league/iso3166": "^4.0",
```
Comment (SamMousa) [inline composer.json:50]: Why is this upgraded? This seems be unrelated to psalm.

**Comment by SamMousa** [comment]
Psalm should run in CI as well.

**Review by SamMousa — COMMENTED** [review]
Conflicts need to be resolved.

**Comment by SamMousa** [comment]
https://psalm.dev/docs/running_psalm/dealing_with_code_issues/#using-a-baseline-file

**Comment by pgichuki** [comment]
> Initialization should not be needed by everyone, just one person does it and you commit the shared configuration to the repo.

Sure. Was for documentation purposes. Editted.

File: composer.json (hunk 1 above)
Comment (pgichuki) [reply composer.json:50]: composer requires this to be updates.

**Pull request closed without merging**